clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy

```diff
@@ -205,6 +208,58 @@ class LoadBalancerV2UpsertHandler {
     return true
   }
 
+  static boolean containsAllRules(List<Rule> aRules, List<Rule> bRules) {
+    !aRules.any { aRule ->
+      boolean foundMatchingRule = bRules.any { bRule ->
+        bRule.actions.containsAll(aRule.actions) && aRule.actions.containsAll(bRule.actions) &&
+          bRule.conditions.containsAll(aRule.conditions) && aRule.conditions.containsAll(bRule.conditions) &&
+          bRule.priority == aRule.priority
+      }
+      return !foundMatchingRule
+    }
+  }
+
+  static void updateListener(String listenerArn,
+                             UpsertAmazonLoadBalancerV2Description.Listener listener,
+                             List<Action> defaultActions, List<Rule> existingRules,
+                             List<Rule> newRules,
+                             AmazonElasticLoadBalancing loadBalancing,
+                             List<String> amazonErrors) {
+    try {
+      loadBalancing.modifyListener(new ModifyListenerRequest()
+        .withListenerArn(listenerArn)
+        .withCertificates(listener.certificates)
+        .withSslPolicy(listener.sslPolicy)
+        .withDefaultActions(defaultActions))
+    } catch (AmazonServiceException e) {
+      String exceptionMessage = "Failed to modify listener ${listenerArn} (${listener.port}:${listener.protocol}) - reason: ${e.errorMessage}."
+      task.updateStatus BASE_PHASE, exceptionMessage
+      amazonErrors << exceptionMessage
+    }
+
+    // Compare the old rules; if any are different, just replace them all.
+    boolean rulesSame = existingRules.size() == newRules.size() &&
+      containsAllRules(existingRules, newRules) &&
+      containsAllRules(newRules, existingRules)
+
+    if (!rulesSame) {
+      existingRules.each { rule ->
+        try {
+          loadBalancing.deleteRule(new DeleteRuleRequest(ruleArn: rule.ruleArn))
+        } catch (AmazonServiceException ignore) {}
```

Comment:
Makes sense to swallow the exception, but would like to at least have a comment in the catch body explaining why we don't care.